edge/pkg/servicebus/servicebus.go

```diff
@@ -1,48 +1,77 @@
 package servicebus
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 	"strings"
+	"sync/atomic"
 	"time"
 
+	"github.com/astaxie/beego/orm"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/klog/v2"
 
 	"github.com/kubeedge/beehive/pkg/core"
 	beehiveContext "github.com/kubeedge/beehive/pkg/core/context"
 	beehiveModel "github.com/kubeedge/beehive/pkg/core/model"
 	commonType "github.com/kubeedge/kubeedge/common/types"
 	"github.com/kubeedge/kubeedge/edge/pkg/common/modules"
-	serviceConfig "github.com/kubeedge/kubeedge/edge/pkg/servicebus/config"
+	servicebusConfig "github.com/kubeedge/kubeedge/edge/pkg/servicebus/config"
+	"github.com/kubeedge/kubeedge/edge/pkg/servicebus/dao"
 	"github.com/kubeedge/kubeedge/edge/pkg/servicebus/util"
 	"github.com/kubeedge/kubeedge/pkg/apis/componentconfig/edgecore/v1alpha1"
 )
 
+var (
+	inited int32
+	c      = make(chan struct{})
+)
+
 const (
 	sourceType  = "router_servicebus"
 	maxBodySize = 5 * 1e6
 )
 
 // servicebus struct
 type servicebus struct {
-	enable bool
+	enable  bool
+	server  string
+	port    int
+	timeout int
+}
+
+type ServerRequest struct {
+	Method    string `json:"method"`
+	TargetURL string `json:"targetURL"`
+	Payload   string `json:"payload"`
+}
+
+type ServerResponse struct {
```

Comment:
sry, I don't understand two structures, I only find where we process two structures. 
can you explain where we construct these structures,

Reply:
i know

Reply:
we construct these two structures in buildBasicHandler func to handler request from other application or send response to other application.servicebus become a universal http server to handler any call from any other application